### PR TITLE
fix: Consider allowed globals when they're accessed via globalThis

### PR DESCRIPTION
**Pull request by flovogt**
Fixes `globalThis` usage reported in https://github.com/SAP/ui5-linter/issues/125. 

`top`, `parent` and top level `this` are still not reported but test cases are added.

### CODE REVIEW

File: src/linter/ui5Types/SourceFileLinter.ts [hunk 1]
```diff
@@ -367,6 +367,13 @@ export default class SourceFileLinter {
 		}
 	}
 
+	isWindow(nodetype: string) {
+		return [
+			"Window & typeof globalThis",
+			"typeof globalThis",
+		].some((v) => nodetype === v);
```
Comment (RandomByte) [inline src/linter/ui5Types/SourceFileLinter.ts:374]:
```suggestion
		].includes(nodetype);
```

Comment (flovogt) [reply src/linter/ui5Types/SourceFileLinter.ts:374]: done

**Review by RandomByte — DISMISSED** [review]

File: src/linter/ui5Types/SourceFileLinter.ts [hunk 2]
```diff
@@ -367,6 +367,14 @@ export default class SourceFileLinter {
 		}
 	}
 
+	isGlobalThis(nodeType: string) {
+		return [
+			"Window & typeof globalThis",
+			"typeof globalThis",
```
Comment (flovogt) [inline src/linter/ui5Types/SourceFileLinter.ts:373]: check which value node type has in case of `self`

Comment (flovogt) [reply src/linter/ui5Types/SourceFileLinter.ts:373]: its `Window & typeof globalThis`

File: test/lib/linter/rules/snapshots/NoGlobals.ts.md [hunk 3]
```diff
@@ -45,28 +57,52 @@ Generated by [AVA](https://avajs.dev).
...
+            category: 1,
+            column: 4,
+            line: 19,
+            message: 'Unable to analyze this method call because the type of identifier "requireSync" in "parent.sap.ui.requireSync()"" could not be determined',
           },
           {
             category: 1,
             column: 4,
             line: 20,
+            message: 'Unable to analyze this method call because the type of identifier "requireSync" in "self.sap.ui.requireSync()"" could not be determined',
```
[the remark this answers is not in the document]
Comment (flovogt) [reply test/lib/linter/rules/snapshots/NoGlobals.ts.md:72]: This difference is, that now `self` is detected the same way as `window`, https://github.com/SAP/ui5-linter/blob/cc2dd31e2c7473fab7856c53a8167455d3a45671/test/lib/linter/rules/snapshots/NoGlobals.ts.md?plain=1#L48.
So behavior is the same

**Review by RandomByte — APPROVED** [review]

**Pull request merged**